### PR TITLE
Add a test contract with tests for it from nomic lab's hardhat boilerplate

### PR DESCRIPTION
**Pull request by johnnylambada**

### CODE REVIEW

**Review by njelich — APPROVED** [review]

**Review by njelich — APPROVED** [review]

**Comment by austintgriffith** [comment]
I think it's a great idea to have a better test example for scaffold-eth 
I don't want to mess with the existing contract on the master branch though. 
This contract is nice and simple and makes onboarding smooth. 

I think we should set this branch up as a testing example and we should link to it in the repo. 

I think we should also spend a little time on test that exists on master and make sure that you can test the "setPurpose" contract. 

I'm down to make the edits to the master scaffold readme that teach you how to run the basic test (testing is something obvious that is missing from scaffold-eth) 

So TL;DR -- Can we make this it's own branch, then make a super simple test on master, then set the readme up to tell the user how to test their "setPurpose" contract, but also set the readme up to link to the token test branch to learn more about testing?

**Comment by codenamejason** [comment]
@johnnylambada I can create a new branch for you to push to also, let me know when you have some time and we can chat about this with @austintgriffith about adding some tests to master as well.

**Comment by johnnylambada** [comment]
@codenamejason @austintgriffith I'm down to do both. For now I'll update this PR to just test the existing contract and remove the new contract and tests from it. I'm also down to work the bulk of this PR into a new `scaffold-eth-testing` (or something like that) branch.

**Comment by codenamejason** [comment]
@johnnylambada any updates

**Comment by johnnylambada** [comment]
@codenamejason sorry! I got completely sidetracked by another project. I'll try to get to this soon :)
